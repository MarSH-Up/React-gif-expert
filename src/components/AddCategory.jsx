import { useState } from "react";
import PropTypes from 'prop-types'

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState(' ');

    const onInputChange = ({target}) => {
        setInputValue(target.value);
    };
    const onSubmit = (event) => {
        event.preventDefault();
        let receivedValue = inputValue.trim();
        if(receivedValue.length <= 1) return;

        onNewCategory(receivedValue);
        setInputValue('');
        //setCategories(categories => [...categories, inputValue])
    };
  return (
    <form onSubmit={onSubmit} aria-label='form'>
        <input
            type="text"
            placeholder="Search gifs"
            value={inputValue}
            onChange={onInputChange }
        />

    </form>
  );
};

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired, 
}