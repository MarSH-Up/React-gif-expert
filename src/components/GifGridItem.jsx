import PropTypes from 'prop-types'

export const GifGridItem = ({id, title, url}) => {
  return (
    <div className='card'>
        <img src={url} className="card" alt={title}/>
        <p key={id}>{title.toUpperCase()}</p>
    </div>
  )
}

GifGridItem.propTypes ={
  title: PropTypes.string.isRequired, 
  url: PropTypes.string.isRequired,
}
