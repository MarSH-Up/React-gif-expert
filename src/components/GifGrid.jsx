import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({item}) => { 

    const {images, isLoading} = useFetchGifs(item);
    return (
        <>
            <h3>{item}</h3>
            {
                isLoading && (<h2 >Loading...</h2>)
            }
            
            <div className="card-grid">
                {
                    images.map( (image) =>(
                        <GifGridItem
                            key={image.id}
                            {...image}
                        />
                    ))
                }
            </div>
        </>
      )
}
