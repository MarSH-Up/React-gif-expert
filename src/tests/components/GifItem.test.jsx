import { render, screen} from "@testing-library/react"
import { GifGridItem } from "../../components/GifGridItem"

describe('Testing GifItem',() =>{
    const title = 'Saitama';
    const url = 'http://localhost/www.onepunchman.com/saitama.jpg';
    test('Matching SnapShot', () =>{
        
        const {container} = render( <GifGridItem title={title} url={url}/> )
        expect(container).toMatchSnapshot();
    });
    test('Showing URL and correct ALT', () =>{
        render( <GifGridItem title={title} url={url}/> )
        //screen.debug();
        //expect( screen.getByRole('img').src).toBe(url);
   
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Confirming that show the title', () =>{
        render( <GifGridItem title={title} url={url}/>);
        expect(screen.getByText(title.toUpperCase())).toBeTruthy(); //Confirming if exist
    })
})