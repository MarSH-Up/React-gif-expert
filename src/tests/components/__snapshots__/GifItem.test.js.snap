// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifItem Matching SnapShot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      class="card"
      src="http://localhost/www.onepunchman.com/saitama.jpg"
    />
    <p>
      SAITAMA
    </p>
  </div>
</div>
`;
